### PR TITLE
Graph Profiler: save() and load() Functionality

### PR DESCRIPTION
**Pull request by micdavis**
- Added functionality for `save()` and `load()` for the `GraphProfiler`
- Wrote a test to cover this new functionality

### CODE REVIEW

File: dataprofiler/profilers/graph_profiler.py [hunk 1]
```diff
@@ -390,3 +392,74 @@ def _attribute_data_as_list(self, graph, attribute):
...
+        :param filepath: Path of file to load from
+        :type filepath: String
+        :return: GraphProfiler being loaded
+        :rtype: GraphProfiler
+        """
+        # Load profile from disk
+        with open(filepath, "rb") as infile:
+            data = pickle.load(infile)
+
+        profiler = cls(None, options=None)
```
Comment (taylorfturner) [inline dataprofiler/profilers/graph_profiler.py:460]: just any class? should there not be some logic around loading the correct class for the profiler?

File: dataprofiler/profilers/graph_profiler.py [hunk 2]
```diff
@@ -1,9 +1,11 @@
 """Class and functions to calculate and profile properties of graph data."""
 from collections import defaultdict
+from datetime import datetime
 
 import networkx as nx
 import numpy as np
 import pandas as pd
+import pickle
```
Comment (JGSweets) [inline dataprofiler/profilers/graph_profiler.py:8]: pickle is built-in so I think should be within the top imports

File: dataprofiler/profilers/graph_profiler.py [hunk 3]
```diff
@@ -390,3 +392,74 @@ def _attribute_data_as_list(self, graph, attribute):
...
+            "_num_nodes": self._num_nodes,
+            "_num_edges": self._num_edges,
+            "_categorical_attributes": self._categorical_attributes,
+            "_continuous_attributes": self._continuous_attributes,
+            "_avg_node_degree": self._avg_node_degree,
+            "_global_max_component_size": self._global_max_component_size,
+            "_continuous_distribution": self._continuous_distribution,
+            "_categorical_distribution": self._categorical_distribution,
+            "metadata": self.metadata,
+            "__calculations": self.__calculations,
```
Comment (JGSweets) [inline dataprofiler/profilers/graph_profiler.py:441]: Do we need __calculations bc we aren't saving options, is that the reason?

File: dataprofiler/tests/profilers/test_graph_profiler.py [hunk 4]
```diff
@@ -122,6 +135,40 @@ def test_graph_data_object(self):
...
+        with mock.patch("builtins.open") as m:
+            mock_file = setup_save_mock_open(m)
+            save_profile.save()
+
+            mock_file.seek(0)
+            load_profile = dp.GraphProfiler.load("mock.pkl")
+
+        # Removed to avoid dict equality ambiguity
+        save_profile.profile["continuous_distribution"]["weight"].pop("properties")
+        load_profile.profile["continuous_distribution"]["weight"].pop("properties")
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/test_graph_profiler.py:153]: We should also assert these props are the same.
e.g. above:
```python
save_continuous_distribution_props = profile.profile["continuous_distribution"]["weight"].pop("properties")

# Then can  (unsure if it is a dict or list or what format.
self.assertEqual(save_continuous_distribution_props, load_continuous_distribution_props)

File: dataprofiler/profilers/graph_profiler.py (hunk 3 above)
Comment (taylorfturner) [reply dataprofiler/profilers/graph_profiler.py:441]: don't see options used anywhere...

Comment (micdavis) [reply dataprofiler/profilers/graph_profiler.py:441]: good call, we don't

File: examples/great_expecations/README.md [hunk 5]
```diff
@@ -2,17 +2,17 @@
 [Great Expectations](https://greatexpectations.io/) is an open source tool built around the idea that should always know what to expect from your data. It helps data teams eliminate pipeline debt, through data testing, documentation, and profiling.
```
Comment (micdavis) [inline examples/great_expecations/README.md:2]: pre-commit formatting

File: examples/great_expecations/ge_expect_column_value_confidence_for_data_label.ipynb [hunk 6]
```diff
@@ -183,4 +183,4 @@
  },
```
Comment (micdavis) [inline examples/great_expecations/ge_expect_column_value_confidence_for_data_label.ipynb:183]: pre-commit formatting

File: examples/great_expecations/ge_expect_column_values_vs_profile_min.ipynb [hunk 7]
```diff
@@ -291,4 +291,4 @@
  },
```
Comment (micdavis) [inline examples/great_expecations/ge_expect_column_values_vs_profile_min.ipynb:291]: pre-commit formatting

File: examples/great_expecations/ge_expect_profile_numeric_columns_diff_in_range.ipynb [hunk 8]
```diff
@@ -300,4 +300,4 @@
  },
```
Comment (micdavis) [inline examples/great_expecations/ge_expect_profile_numeric_columns_diff_in_range.ipynb:300]: pre-commit formatting

File: examples/great_expecations/ge_expect_profile_numeric_columns_percent_diff_in_range.ipynb [hunk 9]
```diff
@@ -300,4 +300,4 @@
  },
```
Comment (micdavis) [inline examples/great_expecations/ge_expect_profile_numeric_columns_percent_diff_in_range.ipynb:300]: pre-commit formatting

File: dataprofiler/profilers/graph_profiler.py [hunk 10]
```diff
@@ -23,7 +25,6 @@ def __init__(self, data, options=None):
         Initialize Graph Profiler.
 
         :param data: data
```
Comment (taylorfturner) [inline dataprofiler/profilers/graph_profiler.py:27]:
```suggestion
        :param data: data
        :type name: String
```

**Review by taylorfturner — CHANGES_REQUESTED** [review]

**Review by taylorfturner — APPROVED** [review]

File: dataprofiler/profilers/graph_profiler.py (hunk 1 above)
Comment (JGSweets) [reply dataprofiler/profilers/graph_profiler.py]: cls is the class it is loading within, this is same logic in BaseProfiler which will allow us to merge all 3 profilers in the long term.

File: dataprofiler/profilers/graph_profiler.py (hunk 10 above)
Comment (JGSweets) [reply dataprofiler/profilers/graph_profiler.py:27]: I think this should be: one of these two types:
`data_readers.graph_data.GraphData or nx.Graph`

**Review by JGSweets — CHANGES_REQUESTED** [review]

**Review by taylorfturner — APPROVED** [review]

**Review by JGSweets — APPROVED** [review]

**Pull request merged**